src/OrleansAWSUtils/Providers/Storage/DynamoDBStorage.cs

```diff
@@ -0,0 +1,415 @@
+﻿using Amazon.DynamoDBv2.Model;
+using Newtonsoft.Json;
+using Orleans.Providers;
+using Orleans.Runtime;
+using Orleans.Serialization;
+using OrleansAWSUtils;
+using OrleansAWSUtils.Storage;
+using System;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Orleans.Storage
+{
+    /// <summary>
+    /// AWS DynamoDB basic Storage provider
+    /// </summary>
+    public class DynamoDBStorage : IStorageProvider
+    {
+        internal const string TableNameDefaultValue = "OrleansGrainState";
```

Comment:
a suggestion, move all the configuration parsing to `DynamoDBStorageConfig` in order to separate concerns